### DISCUSSION TITLE
Morphia Quering is not working as expected. version 2.4

### DISCUSSION THREAD

**Discussion by SOCRATESZZ** [MorphiaOrg/morphia#3135] [General]
I have an Entity
```java
@Entity("Book")
class Book{
   @Reference(language_id)
   private Language language;
}

@Entity("Language")
class Language{
   @Id
   public String id;
   // getters and setter
}
```

I am trying to query Books using language Id only

```java
query.filter(Filters.eq("language.id", "1232342342 ==> the ID")).disableValidation().iterator().toList();
```

it is giving an error 
```
dev.morphia.query.ValidationException: Could not resolve path 'language.id' against 'Book'.  Unknown path element: 'language'.
```

**Comment by evanchooly** [comment]
In your query, you'd simply reference the `language` field on your entity and pass the ID value.  You can't "path" down through the referenced type as there are no joins in MongoDB queries.

**Reply by SOCRATESZZ** [reply]
Language get stored as` DB reference`. I Do not have Language Entity reference. I only have `book entity reference, "language" and language Id`.
Complete querying code
```
filterList = [ Filters.eq("language", "12313333--=>id") ];
Query<P> query = this.datastore.find(Book.class);
 for(Filter filter : filterList) {
      query.filter(filter);
   }

 return query.disableValidation().iterator().toList();
```
When i try to execute the query it automatically replaces  the `id` with `null ` from filter->value. Therefore nothing is returned.
